R/obsPlot.R

```diff
@@ -1,274 +1,475 @@
+                                    showBoxplotOriginalData = showPoints,
+                                    theme               = 'CanvasXpress',
+                                    xAxisTitle          = ylab,
+                                    xAxis2Show          = TRUE,
+                                    title               = title,
+                                    showLegend          = FALSE,
+                                    layoutTopology      = paste0(numrow, 'X', numcol),
+                                    segregateSamplesBy  = plotByCol)
         } else {
-            MyPlot <- MyPlot + theme_grey(baseFontSize)
+            plotlist   <- list()
+            plotby_vec <- unique(data[[plotByCol]])
+            obsPlot    <- lapply(plotby_vec, function(x) {
+                data_subset <- data %>%
+                    filter(!!rlang::sym(plotByCol) == x)
+                cx_data <- data_subset %>%
+                    select(!!rlang::sym(valueCol)) %>%
+                    t() %>%
+                    as.data.frame()
+                smp_data <- data_subset %>%
+                    select(-!!rlang::sym(valueCol))
+                rownames(smp_data) <- colnames(cx_data)
+                title <- x
+
+                canvasXpress(data                = cx_data,
+                             smpAnnot            = smp_data,
+                             graphOrientation    = 'vertical',
+                             graphType           = 'Boxplot',
+                             groupingFactors     = groupCol,
+                             boxplotColor        = boxColor,
+                             boxplotDataPointTransparency = boxTransparency,
+                             boxplotMean         = showMean,
+                             boxplotWhiskersType = 'single',
+                             showViolinBoxplot   = violinLayer,
+                             showBoxplotIfViolin = boxLayer,
+                             violinColor         = violinColor,
+                             violinTransparency  = violinTransparency,
+                             smpLabelRotate      = labelAngle,
+                             smpLabelScaleFontFactor = 1,
+                             smpTitle            = xlab,
+                             smpTitleFontStyle   = 'bold',
+                             smpTitleScaleFontFactor = 1,
+                             layoutAdjust        = axisFree,
+                             showBoxplotOriginalData = showPoints,
+                             theme               = 'CanvasXpress',
+                             xAxisTitle          = ylab,
+                             title               = title,
+                             xAxis2Show          = TRUE,
+                             showLegend          = FALSE)
+            })
+        }
+    } else {
+        .addGeoms <- function(obsPlot) {
+            if (boxLayer == TRUE) {
```

Comment:
no need to use comparison operator when we check for boolean value

Reply:
done